server/methods/browseChannels.js

```diff
@@ -24,18 +24,20 @@ const sortUsers = function(field, direction) {
 
 
 Meteor.methods({
-	browseChannels({text='', type = 'channels', sortBy = 'name', sortDirection = 'asc', page = 0, limit = 10}) {
+	browseChannels({text = '', type = 'Channels', sortBy = 'name', sortDirection = 'asc', page = 0, limit = 10}) {
 		const regex = new RegExp(s.trim(s.escapeRegExp(text)), 'i');
-
-		if (!['channels', 'users', 'private'].includes(type)) {
-			return;
+		let roomType;
+		if (type !== 'users') {
+			roomType = Object.keys(RocketChat.roomTypes.roomTypes).find(key => RocketChat.roomTypes.roomTypes[key]._label === type);
```

Comment:
Passing a value for a new `roomType` parameter will be another way to avoid this computation of room type again.